Read/write polling info on redis

frapontillo opened the pull request:
Since docker managed instances are often shut down and restarted, we need a way to persist the "last polled date" information to avoid spamming subscribers.

frapontillo committed:
refactor(ChatRepository): rename to BotRepository

frapontillo committed:
docs(README): fix le syntax

frapontillo committed:
feat(BotRepository): add polling persistance methods

frapontillo committed:
feat(SioFb): use repository to fetch and persist polling information

rock3r commented:
![](https://media.giphy.com/media/xT77XHUkpCtKbmtuH6/giphy.gif)

frapontillo commented:
Let's do this. @Takhion

frapontillo commented:
```
  / \
 /   \
/     \
-------
  | |
  | |
  | |
```

AHAHAHLOL THIS

rock3r commented:
AHAHAHLOL @frapontillo good luck with that

![](https://media.giphy.com/media/B8tlHMyTvKsa4/giphy.gif)

fourlastor commented on lib/botRepository.js:
```diff
@@ -17,17 +18,25 @@ class ChatRepository {
       .flatMap(el => el);
   }
 
+  getLastPolledDate() {
```
:-1: would have split them in 2 repositories (you even use those 2 methods only in another class)

frapontillo replied:
IDK, splitting in two repos would naturally lead to a super class that will simply handle the `redis` construction just to have a couple of methods...

fourlastor replied:
I don't see the big deal in having a shared collaborator instance to be honest, but I see your point since this project isn't going to grow much more than the current status.

Up to you, not a blocker

fourlastor commented on lib/botRepository.js:
```diff
@@ -17,17 +18,25 @@ class ChatRepository {
   addChat(chatId) {
     this.redis.sadd(CHATS_SET_KEY, chatId);
-  };
+  }
```
lol @ me for not knowing how to JS :dancer:

frapontillo replied:
LOL @ me as well https://github.com/lecoselol/node-telegram-sio-bot/blob/develop/lib%2FsioBot.js#L72

fourlastor commented on lib/sioFb.js:
```diff
@@ -49,13 +51,18 @@ class SioFb {
+      .selectMany(date => {
+        console.log('Polling last run on %s.', date);
+        lastPolledDate = date;
+        return this.getLatestStrips();
```
if you move the filter here you don't  need to scope `lastPolledDate` outside and it reads a bit nicer:

eg:

`.selectMany(date => getStripsAfter(date))`

getStripAfter =>

`return this.getLatestStrips().filter(... here filter by date...)`

fourlastor replied:
also this feels like it's a private method, you can achieve it declaring it as a `function()` inside the scope and you won't have it exposed.

Not sure if that's still a thing in es6, it was when you had to declare classes in es5

frapontillo replied:
Sounds much cleaner, will do.

frapontillo committed:
style(sioFb): get rid of semicolons

frapontillo committed:
refactor(sioFb): move filtering param outside of scope

frapontillo committed:
fix(sioFb): sort pics from graph api

Transform `created_time` property into `Date` object
Manually fix the stupid Facebook ordering w/ `Array.prototype.sort`

frapontillo commented:
@fourlastor please check the latest diffs

![bananoes](http://i.giphy.com/ua4M25Q2HWqXK.gif)

frapontillo commented on lib/sioFb.js:
```diff
@@ -13,6 +13,14 @@ const PHOTOS_CONFIG = {
+  console.log('Polling last run on %s.', lastPolledDate);
+  return this.getLatestStrips()
+    .filter(x => {
+      return (lastPolledDate === null || x.created_time > lastPolledDate);
```
Of course this works only because of https://github.com/lecoselol/node-telegram-sio-bot/pull/2/commits/f44d04c2b14f60d39f8412191181c0481d6a7cd9#diff-0f2305cda36e5ca1efec451070815ae4R52.

Many lols

fourlastor commented:
Seems good now!

frapontillo commented:
You know what to do

![shipit](http://i.giphy.com/ZnUzBMyINSjT2.gif)

rock3r commented:
![](https://media0.giphy.com/media/aKJISWhNCEBxK/200w.gif)
